src/surface/surface_mesh.cpp

```diff
@@ -844,6 +844,11 @@ bool SurfaceMesh::flip(Edge eFlip) {
   Vertex vb = hb1.vertex();
   Vertex vc = ha3.vertex();
   Vertex vd = hb3.vertex();
+  // if flipped, make sure it is not a duplicate
+  for(Vertex v : vc.adjacentVertices()){
+    if(v == vd) return false;
+  }
+  
```

Comment:
I totally understand how you would probably want to avoid self-edges. But there are some situations where they are desired! In particular, I have a personal vested interest in this change because changing the default behavior like this would break mountains of my intrinsic triangulation code :) 

Can we at least guard this check behind a flag like `flip(preventSelfEdge=false)`?

Reply:
Yep, added the optional argument `bool preventSelfEdges = false` to `flip`.